app/src/main/java/fr/free/nrw/commons/filepicker/PickedFiles.java

```diff
@@ -157,11 +153,12 @@ static void scanCopiedImages(Context context, List<File> copiedImages) {
      * @return Uploadable file ready for tag redaction.
      */
     public static UploadableFile pickedExistingPicture(@NonNull Context context, Uri photoUri) throws IOException, SecurityException {// SecurityException for those file providers who share URI but forget to grant necessary permissions
-        InputStream pictureInputStream = context.getContentResolver().openInputStream(photoUri);
         File directory = tempImageDirectory(context);
         File photoFile = new File(directory, UUID.randomUUID().toString() + "." + getMimeType(context, photoUri));
         if (photoFile.createNewFile()) {
+            InputStream pictureInputStream = context.getContentResolver().openInputStream(photoUri);
             writeToFile(pictureInputStream, photoFile);
+            pictureInputStream.close();
```

Comment:
Could you kindly convert this into have a try-with-resouces block like other places so that there's a guarantee that the close happens on input stream even when some exception happens in `writeToFile` method

Reply:
nice catch

Reply:
PR is updated